sourcegraph/learn
# Add bundle analyzer and update React imports

### What should this PR do?
This PR resolves [DEVED-106](https://sourcegraph.atlassian.net/browse/DEVED-106) by adding a bundle analyzer to our project and updating some of our import statements.

You can now run `npm run analyze` to look at the current state of the app. It will be helpful to also use tools like https://bundlephobia.com/ before adding any new packages to the app.

This PR also updates our React imports to work with [current thinking](https://github.com/vercel/next.js/issues/12964#issuecomment-629642157), avoiding redundant imports. See also [this PR](https://github.com/vercel/next.js/pull/13907)

### Why are we making this change?
We want our user experience to remain optimal.

### What are the acceptance criteria? 
- The app works as expected.
- The analyzer works as expected.

### How should this PR be tested?
- Check out the branch, nuke `node_modules`, and install dependencies with `npm ci`.
- Run `npm run analyze` and ensure that analyzer works.
- Run the app and check for unexpected behavior.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
